### PR TITLE
legal: fix licenses of dependencies

### PR DESCRIPTION
**Pull request by florianl**
Replace the costum license checker with google/go-licenses, which is already used in other OTel projects. And run the license check for dependencies as part of the CI.

This fix also makes sure, that the licenses are Go module version aware.

### CODE REVIEW

File: legal/non-go-dependencies.json [hunk 1]
```diff
@@ -1,7 +1,9 @@
...
-}
+[
+	{
+		"Dependency": "zyantific/zydis",
+		"Version": "v3.1.0",
+		"Licence": "MIT",
+		"URL": "https://zydis.re",
+		"LicenceFile": "https://raw.githubusercontent.com/zyantific/zydis/v3.1.0/LICENSE"
+	}
+]
```
Comment (athre0z) [inline legal/non-go-dependencies.json:9]: Not actually your mistake (I forgot to update this), but can we fix this while we are changing this anyway?

```suggestion
[
	{
		"Dependency": "zyantific/zydis",
		"Version": "v4.1.0",
		"Licence": "MIT",
		"URL": "https://zydis.re",
		"LicenceFile": "https://raw.githubusercontent.com/zyantific/zydis/v4.1.0/LICENSE"
	}
]
```

Comment (florianl) [reply legal/non-go-dependencies.json]: sure - makes sense :+1:

**Comment by florianl** [comment]
With https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/143/commits/f8f8db834178d046a88ce7d266614d2de3e30228 applied, CI is failing as expected, as the licenses where not updates https://github.com/open-telemetry/opentelemetry-ebpf-profiler/actions/runs/10683289144/job/29611021732?pr=143.

**Comment by rockdaboot** [comment]
:+1: Thanks

**Pull request merged**